aws/sqs/main.tf

```diff
@@ -0,0 +1,35 @@
+locals {
+  # This helps avoid queue names ending in "-" or "-.fifo"
+  given_queue_name = var.queue_name == "" ? "" : "-${var.queue_name}"
+  # All fifo queues must end in .fifo, per AWS rules
+  queue_suffix    = var.is_fifo == true ? ".fifo" : ""
+  full_queue_name = "${var.stack}-${var.env}${local.given_queue_name}${local.queue_suffix}"
+}
+
+resource "aws_sqs_queue" "this" {
+  name                        = local.full_queue_name
+  fifo_queue                  = var.is_fifo
+  content_based_deduplication = var.content_based_deduplication
+  receive_wait_time_seconds   = var.receive_wait_time_seconds
+  visibility_timeout_seconds  = var.visibility_timeout_seconds
+}
+
+resource "aws_sqs_queue_policy" "this" {
```

Comment:
On a more general note, we probably shouldn't have a default policy, especially one so permissive (this one makes any action on the queue open to anyone in the world i believe). At best, we could have a policy statement variable that can be passed in. For most cases, permission to the queue should be granted through an IAM role rather than a queue policy

Reply:
Good call. I started this a month ago based on what we had in perfected. I had to make a number of minor tweaks, so I'm not surprised the policy isn't perfect. I think cognito does a policy passing, or something close to that. i'll review and see if I can follow the same thing.